all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst

```diff
@@ -0,0 +1,232 @@
+
+   cat > user-data.yaml <<EOF
+   #cloud-config
+   users:
+     - name: root
+           disable_root: false
+           ssh_authorized_keys:
+           - $SSH_PUB_KEY$
+   chpasswd:
+     expire: false
+     users:
+           - name: root
+           password: $PASSWORD$
+           type: text
+   packages:
+     - livecd-rootfs
+   package_update: true
+   package_upgrade: true
+   write_files:
+     - content: |
+           Defaults !use_pty
+           path: /etc/sudoers.d/disable_pty
+   EOF
+
+   cloud-localds seed.img user-data.yaml
+
+.. important::
+   It is necessary to input an ssh key for our final step: getting our built image outside of the build VM. You can learn more about generating ssh keys `here <https://ubuntu.com/server/docs/service-openssh>`_.
+
+Download a cloud image
+~~~~~~~~~~~~~~~~~~~~~~
+You will need a cloud image of the same architecture as the image that you want to build. You can download an image from `Ubuntu Cloud Images <http://cloud-images.ubuntu.com/>`_ and place it in your build folder. Alternatively, you can use ``wget``. For a Ubuntu 22.04 Jammy image, run:
+
+.. code:: bash
+
+   wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+
+The Vagrant box build process uses more disk space than is available by default on the provided cloud images, so we will resize the image using ``qemu-img``. For our image, ``jammy-server-cloudimg-amd64.img``, run:
+
+.. code:: bash
+
+   sudo apt install qemu-utils
+   qemu-img resize jammy-server-cloudimg-amd64.img +10G
+
+This will add 10GB to the virtual disk of our image and give us enough space to complete the Vagrant box build.
+
+Launch a VM with QEMU
+---------------------
+We will use `SeaBIOS <https://www.seabios.org/SeaBIOS>`_ to boot our image. SeaBIOS booting an image is essentially the easiest as it is the default firmware used by QEMU.
+
+The following command launches a cloud image with:
+
+- KVM acceleration
+- the local machine's CPUs
+- 2GB of memory
+- no graphics and serial output to the console
+- snapshot will make writes to a temporary file instead of the disk image itself. This ensures the base disk is not touched. If at some point, you want to persist the changes you've made on the disk, press ``C-a s``
+- virtio network device that redirects guest port 22 to host's port 2222
+- virtio cloud image
+- virtio seed image
+
+.. code:: bash
+
+   qemu-system-x86_64  \
+     -cpu host -machine type=q35,accel=kvm -m 2048 \
+     -nographic \
+     -snapshot \
+     -netdev id=net00,type=user,hostfwd=tcp::2222-:22 \
+     -device virtio-net-pci,netdev=net00 \
+     -drive if=virtio,format=qcow2,file=./jammy-server-cloudimg-amd64.img \
+     -drive if=virtio,format=raw,file=seed.img
+
+.. note::
+   To exit the QEMU shell, press ``C-a x``
+
+Configure the environment
+-------------------------
+All the commands can be run in an ssh session except ``lb build`` that **has to be run via the serial console**. 
```

Comment:
this is interesting, and might warrant a small "why"

Reply:
Removed that document page altogether.